Core/src/Propagator/codegen/sympy_stepper_math.hpp

```diff
@@ -0,0 +1,349 @@
+  dGdTL[9] = x33 * x46 + x33 * dk4dTL[9] + x57 * dk2dTL[9] + x57 * dk3dTL[9];
+  dGdTL[10] =
+      x33 * x43 + x33 * dk4dTL[10] + x57 * dk2dTL[10] + x57 * dk3dTL[10];
+  dGdTL[11] =
+      x33 * x41 + x33 * dk4dTL[11] + x57 * dk2dTL[11] + x57 * dk3dTL[11];
+  T new_J[64];
+  new_J[0] = 1;
+  new_J[1] = 0;
+  new_J[2] = 0;
+  new_J[3] = 0;
+  new_J[4] = 0;
+  new_J[5] = 0;
+  new_J[6] = 0;
+  new_J[7] = 0;
+  new_J[8] = 0;
+  new_J[9] = 1;
+  new_J[10] = 0;
+  new_J[11] = 0;
+  new_J[12] = 0;
+  new_J[13] = 0;
+  new_J[14] = 0;
+  new_J[15] = 0;
+  new_J[16] = 0;
+  new_J[17] = 0;
+  new_J[18] = 1;
+  new_J[19] = 0;
+  new_J[20] = 0;
+  new_J[21] = 0;
+  new_J[22] = 0;
+  new_J[23] = 0;
+  new_J[24] = 0;
+  new_J[25] = 0;
+  new_J[26] = 0;
+  new_J[27] = 1;
+  new_J[28] = 0;
+  new_J[29] = 0;
+  new_J[30] = 0;
+  new_J[31] = 0;
+  new_J[32] = J[32] * dGdTL[0] + J[40] * dGdTL[1] + J[48] * dGdTL[2] + dFdTL[0];
+  new_J[33] = J[33] * dGdTL[0] + J[41] * dGdTL[1] + J[49] * dGdTL[2] + dFdTL[1];
+  new_J[34] = J[34] * dGdTL[0] + J[42] * dGdTL[1] + J[50] * dGdTL[2] + dFdTL[2];
+  new_J[35] = 0;
+  new_J[36] = J[36] * dGdTL[0] + J[44] * dGdTL[1] + J[52] * dGdTL[2];
+  new_J[37] = J[37] * dGdTL[0] + J[45] * dGdTL[1] + J[53] * dGdTL[2];
+  new_J[38] = J[38] * dGdTL[0] + J[46] * dGdTL[1] + J[54] * dGdTL[2];
+  new_J[39] = 0;
+  new_J[40] = J[32] * dGdTL[3] + J[40] * dGdTL[4] + J[48] * dGdTL[5] + dFdTL[3];
+  new_J[41] = J[33] * dGdTL[3] + J[41] * dGdTL[4] + J[49] * dGdTL[5] + dFdTL[4];
+  new_J[42] = J[34] * dGdTL[3] + J[42] * dGdTL[4] + J[50] * dGdTL[5] + dFdTL[5];
+  new_J[43] = 0;
+  new_J[44] = J[36] * dGdTL[3] + J[44] * dGdTL[4] + J[52] * dGdTL[5];
+  new_J[45] = J[37] * dGdTL[3] + J[45] * dGdTL[4] + J[53] * dGdTL[5];
+  new_J[46] = J[38] * dGdTL[3] + J[46] * dGdTL[4] + J[54] * dGdTL[5];
+  new_J[47] = 0;
+  new_J[48] = J[32] * dGdTL[6] + J[40] * dGdTL[7] + J[48] * dGdTL[8] + dFdTL[6];
+  new_J[49] = J[33] * dGdTL[6] + J[41] * dGdTL[7] + J[49] * dGdTL[8] + dFdTL[7];
+  new_J[50] = J[34] * dGdTL[6] + J[42] * dGdTL[7] + J[50] * dGdTL[8] + dFdTL[8];
+  new_J[51] = 0;
+  new_J[52] = J[36] * dGdTL[6] + J[44] * dGdTL[7] + J[52] * dGdTL[8];
+  new_J[53] = J[37] * dGdTL[6] + J[45] * dGdTL[7] + J[53] * dGdTL[8];
+  new_J[54] = J[38] * dGdTL[6] + J[46] * dGdTL[7] + J[54] * dGdTL[8];
+  new_J[55] = 0;
+  new_J[56] = J[32] * dGdTL[9] + J[40] * dGdTL[10] + J[48] * dGdTL[11] + J[56] +
+              dFdTL[9];
+  new_J[57] = J[33] * dGdTL[9] + J[41] * dGdTL[10] + J[49] * dGdTL[11] + J[57] +
+              dFdTL[10];
+  new_J[58] = J[34] * dGdTL[9] + J[42] * dGdTL[10] + J[50] * dGdTL[11] + J[58] +
+              dFdTL[11];
+  new_J[59] = J[59] + h * lambda * x35 / dtds;
+  new_J[60] = J[36] * dGdTL[9] + J[44] * dGdTL[10] + J[52] * dGdTL[11] + J[60];
+  new_J[61] = J[37] * dGdTL[9] + J[45] * dGdTL[10] + J[53] * dGdTL[11] + J[61];
+  new_J[62] = J[38] * dGdTL[9] + J[46] * dGdTL[10] + J[54] * dGdTL[11] + J[62];
+  new_J[63] = 1;
+  J[0] = new_J[0];
```

Comment:
Are they continuous in memory, ie. could this one memory copy of a block ?

Reply:
I am also not happy with this assignment. I played a bit with it to see if it has any performance implications and it seems like it doesn't. At the moment this is code that is directly generated through sympy but we could overload the behavior with `std::copy` in the future to make it more concise

Reply:
Ok, sounds good.